### ISSUE TITLE
IP Leaks of default firefox settings on temporary proxy failure (auto fallback to firefox settings)

### ISSUE THREAD

**Issue by firefox4ever** [bekh6ex/firefox-container-proxy#55]
It doesn't an issue of your add-on. Just firefox proxy API is broken :( And as mozilla developers say - it's a feature, not a bug.

More info:
https://bugzilla.mozilla.org/show_bug.cgi?id=1750561

**Comment by numb699** [comment]
Too bad, kinda makes a fantastic add-on useless.

**Comment by firefox4ever** [comment]
No, add-on works well, but except some situations

**Comment by numb699** [comment]
Yes I agree. My situation being using a proxy ( Many proxies in containers) without risking DNS leaks.  For me that's the whole point and the single most important use case. Now I 'm back to full blown VPN or other more inconvenient means of proxying.

**Comment by firefox4ever** [comment]
I found this bug because I'm developing own add-on for proxy per container binding.
And a little chance that Mozilla developers will fix this bug still exist, because bug is still open now.

**Comment by firefox4ever** [comment]
> Yes I agree. My situation being using a proxy ( Many proxies in containers) without risking DNS leaks. For me that's the whole point and the single most important use case. Now I 'm back to full blown VPN or other more inconvenient means of proxying.

You can comment about this issue in Bugzilla:
https://bugzilla.mozilla.org/show_bug.cgi?id=1750561

**Comment by firefox4ever** [comment]
OMG, bug on bugzilla is closed - not a bug :(
I will try to use a suggested in bugzilla bug report code in my add-on and will write result here (is it work or not)

**Comment by firefox4ever** [comment]
I tested fix proposed in Mozilla bugtracker (undocumented feature). Returning `[{ ... proxy info ...} , null]` stops fallback to firefox proxy settings. So bug can be fixed on add-on side

**Comment by firefox4ever** [comment]
@numb699 you can use my addon until bug be fixed here (but it limited only to socks proxies).
https://addons.mozilla.org/ru/firefox/addon/containers-proxy-helper/
